pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -737,4 +720,36 @@ private static Mode convertRegexSubscriptionMode(RegexSubscriptionMode regexSubs
             return null;
         }
     }
+
+    private SchemaInfoProvider loadSchemaProvider(String topicName) {
+        return new MultiVersionSchemaInfoProvider(TopicName.get(topicName), this);
+    }
+
+    public LoadingCache<String, SchemaInfoProvider> getSchemaProviderLoadingCache() {
+        return schemaProviderLoadingCache;
+    }
+
+    private void schemaHandle(Schema schema, String topicName) {
```

Comment:
`schemaHandle` is not an ambiguous name. I would suggest calling it `preProcessSchemaBeforeSubscribe`, which indicates the method is called before subscribe or createReader.